File: BE/Hinet.Model/Entities/TuyenDung/TD_TuyenDung.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hinet.Model.Entities.TuyenDung
{
    public class TD_TuyenDung:AuditableEntity
    {
        [StringLength(250)]
        public string TenViTri { get; set; }
        public Guid? PhongBanId { get; set; }
        public int SoLuongCanTuyen { get; set; }
        public DateOnly NgayBatDau { get; set; }
        public DateOnly NgayKetThuc { get; set; }
        public string? MoTa { get; set; }
        public TinhTrang_TuyenDung TinhTrang { get; set; } = 0;
        public Loai_TuyenDung Loai { get; set; } = 0;
        public HinhThuc_TuyenDung HinhThuc { get; set; } = 0;
    }
    public enum TinhTrang_TuyenDung
    {
        DangTuyen,
        DaDong,
        Hoan
    }
     public enum Loai_TuyenDung
    {
        ThucTap,
        NhanVien,
        Hoan
    }
     public enum HinhThuc_TuyenDung
    {
        FullTime,
        PartTime
    }
}

File: FE/src/types/TD_TuyenDung/TD_TuyenDung.ts
import dayjs from "dayjs";
import { EntityType, SearchBase } from "../general";

export interface TD_TuyenDungType extends EntityType {
  tenViTri: string;
  phongBanId?: string;
  soLuongCanTuyen: number;
  ngayBatDau: string | dayjs.Dayjs;
  ngayKetThuc: string | dayjs.Dayjs;
  moTa?: string;
  tinhTrang: number;
  loai?: number;
  hinhThuc?: number;
  // Thêm field từ DTO
  tenPhongBan?: string;
  // Audit fields
  createdDate?: string | dayjs.Dayjs;
  createdBy?: string;
  modifiedDate?: string | dayjs.Dayjs;
  modifiedBy?: string;
}

export interface TD_TuyenDungSearchType extends SearchBase {
  tenViTri?: string;
  phongBanId?: string;
  soLuongCanTuyen?: string;
  ngayBatDau?: dayjs.Dayjs | null;
  ngayKetThuc?: dayjs.Dayjs | null;
  tinhTrang?: number;
}

export interface TD_TuyenDungCreateType {
  tenViTri: string;
  phongBanId?: string;
  soLuongCanTuyen: string;
  ngayBatDau: dayjs.Dayjs;
  ngayKetThuc: dayjs.Dayjs;
  moTa?: string;
  tinhTrang: string;
  loai: number;
  hinhThuc: number;
}

export interface TD_TuyenDungEditType extends TD_TuyenDungCreateType {
  id: string;
}

// Constants cho trạng thái
export const TD_TuyenDungTinhTrangOptions = [
  { value: 0, label: "Đang tuyển dụng" },
  { value: 1, label: "Đã đóng" },
  { value: 2, label: "Hoãn" },
];

export const getTinhTrangLabel = (value: number): string => {
  const option = TD_TuyenDungTinhTrangOptions.find(opt => opt.value === value);
  return option?.label || "Không xác định";
};

export enum Loai_TuyenDung {
  ThucTap = 0,
  NhanVien = 1,
  Hoan = 2,
}

export enum HinhThuc_TuyenDung {
  FullTime = 0,
  PartTime = 1,
}

export const TD_TuyenDungLoaiOptions = [
  { value: Loai_TuyenDung.ThucTap, label: "Thực tập" },
  { value: Loai_TuyenDung.NhanVien, label: "Nhân viên" }
];

export const TD_TuyenDungHinhThucOptions = [
  { value: HinhThuc_TuyenDung.FullTime, label: "FullTime" },
  { value: HinhThuc_TuyenDung.PartTime, label: "PartTime" },
];

export const getLoaiLabel = (value?: number): string => {
  const option = TD_TuyenDungLoaiOptions.find(opt => opt.value === value);
  return option?.label || "Không xác định";
};

export const getHinhThucLabel = (value?: number): string => {
  const option = TD_TuyenDungHinhThucOptions.find(opt => opt.value === value);
  return option?.label || "Không xác định";
};
